### PR TITLE
Deserialize headers (now with CI)

### PR DESCRIPTION
**Pull request by RikkiGibson**

### CODE REVIEW

File: src/azure/Templates/AzureModelIndexTemplate.cshtml [hunk 1]
```diff
@@ -27,4 +31,14 @@ export { BaseResource, CloudError };
...
+}
+
+@foreach (var method in Model.MethodsWithHeaders) {
+@EmptyLine
+<text>
+/**
+@WrapComment(" * ", $"Contains response data for the {method.Name} operation.")
+ */
+</text>
+@:export interface @(Regex.Replace(method.ReturnType.Headers.Name, "Headers$", "Response")) extends msRest.HttpOperationResponse<@(method.ReturnType.Body?.TSType(inModelsModule: true) ?? "void"), @(method.ReturnType.Headers.Name)> {};
```
Comment (daschult) [inline src/azure/Templates/AzureModelIndexTemplate.cshtml]: I'd recommend moving this line up inside the <text> element so that you don't need the `@:`.

**Comment by RikkiGibson** [comment]
I can install `ms-rest-js@0.7.188` locally but it won't install in CI. Go figure.

**Comment by RikkiGibson** [comment]
# 🤖 AutoRest automatic bundle size check 🤖
## Test bundle size (of all generated code and dependencies): 5.44 MB

**Comment by RikkiGibson** [comment]
# 🤖 AutoRest automatic bundle size check 🤖
## Total size: 5.44 MB

**Comment by RikkiGibson** [comment]
# 🤖 AutoRest automatic bundle size check 🤖
## Total size: 5.44 MB

**Comment by RikkiGibson** [comment]
# 🤖 AutoRest automatic bundle size check 🤖
## Total size: 5.44 MB

**Comment by RikkiGibson** [comment]
# 🤖 AutoRest automatic bundle size check 🤖
## Size before PR: 5.37 MB
## Size after PR: 5.37 MB

**Comment by RikkiGibson** [comment]
# 🤖 AutoRest automatic bundle size check 🤖
### Size increased by 1.37%.
### Size before PR: 5.37 MB
### Size after PR: 5.44 MB

**Comment by RikkiGibson** [comment]
# 🤖 AutoRest automatic bundle size check 🤖
### Size increased by 1.37% (75.39 KB)

**Pull request merged**